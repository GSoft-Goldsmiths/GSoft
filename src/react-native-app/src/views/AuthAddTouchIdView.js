import React from 'react';
import { AsyncStorage, Button, Image, Text, View } from 'react-native';

import { color, fullPageView, style } from '../styles/variables';
import BTN from '../components/Button';

export default class AuthAddTouchIdView extends React.Component {

  constructor(props) {
    super(props);
  }

  _setTouchId = async () => {
    // setup permission
    try {
      const user = { auth: { useTouchId: true } };
      await AsyncStorage.mergeItem('user', JSON.stringify(user));
      this.props.navigation.navigate('AuthFinished');
    } catch (error) {
      console.log(error);
    }
  };

  _skipTouchId = async () => {
    try {
      const user = { auth: { useTouchId: false } };
      await AsyncStorage.mergeItem('user', JSON.stringify(user));
      this.props.navigation.navigate('AuthFinished');
    } catch (error) {
      console.log(error);
    }
  };

  render() {

    const title = 'Just one more step!';
    const subtitle = 'If you forget the passcode, you can\n' +
      ' always use touch ID to unlock it, but first we ' +
      'need your permission to set it up.';
    return (
      <View style={style.container}>
        <Image
          style={fullPageView.image}
          source={require('../../assets/images/icons/fingerprint.png')}
        />
        <Text style={fullPageView.title}>{title}</Text>
        <Text style={fullPageView.subtitle}>{subtitle}</Text>
        <BTN handler={() => this._setTouchId()} text="OK"/>
        <Button
          title="SKIP"
          color={color.main}
          onPress={() => this._skipTouchId()}
        />
      </View>
    );
  }

}
