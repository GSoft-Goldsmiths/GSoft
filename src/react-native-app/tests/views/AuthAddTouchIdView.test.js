import React from 'react';
import renderer from 'react-test-renderer';

import AuthAddTouchIdView from '../../src/views/AuthAddTouchIdView';

it('renders without crashing', () => {
  const rendered = renderer.create(<AuthAddTouchIdView/>).toJSON();
  expect(rendered).toBeTruthy();
});
