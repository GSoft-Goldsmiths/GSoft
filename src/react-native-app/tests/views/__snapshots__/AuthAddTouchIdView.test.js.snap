// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 20,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/icons/fingerprint.png",
      }
    }
    style={
      Object {
        "height": 145,
        "marginBottom": 20,
        "width": 145,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#0093A8",
        "fontFamily": "Caviar Dreams",
        "fontSize": 30,
        "marginBottom": 15,
      }
    }
  >
    Just one more step!
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#0093A8",
        "fontFamily": "Caviar Dreams",
        "fontSize": 15,
        "fontWeight": "500",
      }
    }
  >
    If you forget the passcode, you can
 always use touch ID to unlock it, but first we need your permission to set it up.
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0093A8",
        "height": 36,
        "justifyContent": "center",
        "marginVertical": 20,
        "width": 90,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Caviar Dreams Bold",
        }
      }
    >
      OK
    </Text>
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={undefined}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#0093A8",
            },
          ]
        }
      >
        SKIP
      </Text>
    </View>
  </View>
</View>
`;
